File: csharp/MapleStory.OpenAPI/Src/Dto/Character/CharacterSetEffectSetDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 세트 효과 정보
    /// </summary>
    public class CharacterSetEffectSetDTO
    {
        /// <summary>
        /// 세트 효과 명
        /// </summary>
        [JsonProperty("set_name")]
        public string SetName { get; set; }

        /// <summary>
        /// 세트 개수 (럭키 아이템 포함)
        /// </summary>
        [JsonProperty("total_set_count")]
        public long TotalSetCount { get; set; }

        /// <summary>
        /// 적용 중인 세트 효과 정보
        /// </summary>
        [JsonProperty("set_effect_info")]
        public List<CharacterSetEffectInfoDTO> SetEffectInfo { get; set; }

        /// <summary>
        /// 모든 세트 효과 정보
        /// </summary>
        [JsonProperty("set_option_full")]
        public List<CharacterSetEffectOptionFullDTO> SetOptionFull { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterSetEffectSetDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

/**
 * Set Item effect information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterSetEffectSetDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterSetEffectSetDTO<
        CharacterSetEffectInfoDTO,
        CharacterSetEffectOptionFullDTO> {

    /**
     * Set Item effect name
     */
    @JsonProperty("set_name")
    private String setName;

    /**
     * Number of Set Items (including Lucky Items)
     */
    @JsonProperty("total_set_count")
    private long totalSetCount;

    /**
     * Active Set Item effect information
     */
    @JsonProperty("set_effect_info")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterSetEffectInfoDTO> setEffectInfo;

    /**
     * All Set Item effect information
     */
    @JsonProperty("set_option_full")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterSetEffectOptionFullDTO> setOptionFull;
}
